src/wallet/index.ts

```diff
@@ -81,6 +114,67 @@ export class Wallet {
     this.identityAddresses.push(identity.address)
     return identity
   }
+
+  async createGaiaConfig(gaiaHubUrl: string) {
+    return connectToGaiaHub(gaiaHubUrl, this.configPrivateKey)
+  }
+
+  async fetchConfig(gaiaConfig: GaiaHubConfig): Promise<WalletConfig | null> {
+    try {
+      // await putFile('wallet-config.json', JSON.stringify(this.wall))
+      const response = await fetch(`${gaiaConfig.url_prefix}/${gaiaConfig.address}/wallet-config.json`)
+      const config: WalletConfig = await response.json() 
+      this.walletConfig = config
+      return config
+    } catch (error) {
+      // console.error(error)
+      return null
+    }
+  }
```

Comment:
Why do we need to try/catch here? 
If the promise fails, don't we want it to throw an error, and have the consumer of this function handle the error case, rather than the promise resolving "successful" null?

Reply:
We try/catch because the fetch will fail if the config file isn't found, which is totally normal behavior for this function.

Reply:
Could save some debugging pain later on if this only returned null on 404, and throw on all other conditions